### ISSUE TITLE
Create wazuh-wui packages 

### ISSUE THREAD

**Issue by DFolchA** [wazuh/wazuh-packages#685]
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
|4.2.0 | wui| Install/Upgrade/Remove | RPM/DEB |

Related issue: #665

After the research done in #667 we need to create the tools to generate packages for `wazuh-ui`. This task includes:

- [ ] Generate SPECS for the package
- [ ] Adapt our current tools to generate wazuh-ui package

**Comment by DFolchA** [comment]
### Update Report

- Change OD default start page to wazuh one
- Work on SPECS and start testing packages

**Comment by DFolchA** [comment]
### Update Report
- Work on RPM installation scripts

**Comment by DFolchA** [comment]
### Update Report

Working on RPM SPECS:

1. Create installation scripts to ensure that the service restart after an upgrade and the rest of functionallities that we have in the other Wazuh packages.
https://github.com/wazuh/wazuh-packages/blob/59b53b4a5a362f95279c61b24dcd207c129ebb74/rpms/SPECS/4.3.0/wazuh-wui-4.3.0.spec#L79-L89

2. Added new wazuh-wui service files to repository, and modify the generation tools to use them:
- [init.d](https://github.com/wazuh/wazuh-packages/blob/e4e1eb6f0edf17dd472dab4c71a017dcb91e902c/rpms/services/wazuh-wui) 
- [systemctl](https://github.com/wazuh/wazuh-packages/blob/e4e1eb6f0edf17dd472dab4c71a017dcb91e902c/rpms/services/wazuh-wui.service) 

https://github.com/wazuh/wazuh-packages/blob/59b53b4a5a362f95279c61b24dcd207c129ebb74/rpms/CentOS/6/x86_64/Dockerfile#L65-L66 

3. Work in adding all the files in Wazuh Kibana plugin to the %files section of the specs.

[Wazuh plugin files](https://github.com/wazuh/wazuh-packages/files/6272885/missing_files.txt)

4. Modify default configuration and add certificates.

**Comment by DFolchA** [comment]
### Update Report
- Development moved to 5.0.0 version

### RPM wazuh-wui installation
```
[root@manager ~]# yum install ./wazuh-wui-5.0.0-1.x86_64.rpm 
Complementos cargados:fastestmirror
Examinando ./wazuh-wui-5.0.0-1.x86_64.rpm: wazuh-wui-5.0.0-1.x86_64
Marcando ./wazuh-wui-5.0.0-1.x86_64.rpm para ser instalado
Resolviendo dependencias
--> Ejecutando prueba de transacción
---> Paquete wazuh-wui.x86_64 0:5.0.0-1 debe ser instalado
--> Resolución de dependencias finalizada

Dependencias resueltas

===========================================================================================================================================
 Package                       Arquitectura               Versión                      Repositorio                                   Tamaño
===========================================================================================================================================
Instalando:
 wazuh-wui                     x86_64                     5.0.0-1                      /wazuh-wui-5.0.0-1.x86_64                     697 M

Resumen de la transacción
===========================================================================================================================================
Instalar  1 Paquete

Tamaño total: 697 M
Tamaño instalado: 697 M
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Instalando    : wazuh-wui-5.0.0-1.x86_64                                                                                             1/1 
  Comprobando   : wazuh-wui-5.0.0-1.x86_64                                                                                             1/1 

Instalado:
  wazuh-wui.x86_64 0:5.0.0-1                                                                                                               

¡Listo!
```

### Wazuh WUI service

```
[root@manager ~]# systemctl status wazuh-wui.service 
● wazuh-wui.service - Wazuh-wui
   Loaded: loaded (/etc/systemd/system/wazuh-wui.service; disabled; vendor preset: disabled)
   Active: inactive (dead)
[root@manager ~]# systemctl restart wazuh-wui.service 
[root@manager ~]# systemctl status wazuh-wui.service 
● wazuh-wui.service - Wazuh-wui
   Loaded: loaded (/etc/systemd/system/wazuh-wui.service; disabled; vendor preset: disabled)
   Active: active (running) since jue 2021-04-08 10:42:00 UTC; 42s ago
 Main PID: 11949 (node)
   CGroup: /system.slice/wazuh-wui.service
           └─11949 /usr/share/wazuh-wui/bin/../node/bin/node /usr/share/wazuh-wui/bin/../src/cli/dist -c /etc/wazuh-wui/wazuh-wui.yml

abr 08 10:42:34 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:34Z","tags":[],"pid":11949,"method":"get...) Gecko/
abr 08 10:42:34 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:34Z","tags":[],"pid":11949,"method":"get.../78.0","
abr 08 10:42:34 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:34Z","tags":[],"pid":11949,"method":"get..."accept"
abr 08 10:42:34 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:34Z","tags":[],"pid":11949,"method":"get...fox/78.0
abr 08 10:42:35 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:35Z","tags":[],"pid":11949,"method":"pos...8.0","ac
abr 08 10:42:35 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:35Z","tags":[],"pid":11949,"method":"get...0100101 
abr 08 10:42:35 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:35Z","tags":[],"pid":11949,"method":"get...8.0) Gec
abr 08 10:42:35 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:35Z","tags":[],"pid":11949,"method":"get...fox/78.0
abr 08 10:42:35 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:35Z","tags":[],"pid":11949,"method":"get... rv:78.0
abr 08 10:42:35 manager kibana[11949]: {"type":"response","@timestamp":"2021-04-08T10:42:35Z","tags":[],"pid":11949,"method":"get...1; Ubunt
Hint: Some lines were ellipsized, use -l to show in full.

```

![image](https://user-images.githubusercontent.com/19505384/114013830-449b9600-9868-11eb-92c3-65aa8c533209.png)

**Comment by DFolchA** [comment]
### Update Report

Added first version of Debian SPEC 28de5da2b2a2c251bea223b2c522fd1c7f67fdf8.

**Comment by DFolchA** [comment]
### Update Report

- Investigate errors in `init.d` service:

We found the following errors when running the init.d service for wazuh-wui:

```
# /etc/init.d/wazuh-wui start
wazuh-wui started
[root@65e8171420a1 /]# /etc/init.d/wazuh-wui status
wazuh-wui is not running
[root@65e8171420a1 /]# cat /var/log/wazuh-wui/wazuh-wui.stderr
chroot: cannot change root directory to : No such file or directory
```
Solved in   28de5da

- Test Deb package and apply fixes

**Comment by DFolchA** [comment]
### Update report

Investigate init.d problem:

```
# service wazuh-wui stop
wazuh-wui stop failed; still running.
```

This problem was caused by a wrong configuration of the permission for the directories, and was fixed in this commit: 4f7fe5d 

Work on DEB Packages, Investigate error when installing the Wazuh plugin:

```
/build_wazuh/wui/wazuh-wui-5.0.0
ls -lah //usr/share/wazuh-wui
total 1.2M
drwxr-xr-x   9 wazuh-wui wazuh-wui 4.0K Apr 12 17:03 .
drwxr-xr-x   1 root      root      4.0K Apr 12 17:03 ..
-rw-r--r--   1 wazuh-wui wazuh-wui  12K Jan 13 02:07 LICENSE.txt
-rw-r--r--   1 wazuh-wui wazuh-wui 1.1M Jan 13 02:07 NOTICE.txt
-rw-r--r--   1 wazuh-wui wazuh-wui 3.9K Jan 13 02:07 README.txt
drwxr-xr-x   2 wazuh-wui wazuh-wui 4.0K Mar  2 20:43 bin
drwxr-xr-x   2 wazuh-wui wazuh-wui 4.0K Apr 12 17:03 config
drwxr-xr-x   2 wazuh-wui wazuh-wui 4.0K Jan 13 02:07 data
drwxr-xr-x   6 wazuh-wui wazuh-wui 4.0K Mar  2 20:43 node
drwxr-xr-x 705 wazuh-wui wazuh-wui  20K Mar  2 20:43 node_modules
-rw-r--r--   1 wazuh-wui wazuh-wui  740 Jan 13 02:07 package.json
drwxr-xr-x  10 wazuh-wui wazuh-wui 4.0K Mar  2 20:44 plugins
drwxr-xr-x  11 wazuh-wui wazuh-wui 4.0K Mar  2 20:43 src
ls -lah //usr/share/wazuh-wui/bin/
total 20K
drwxr-xr-x 2 wazuh-wui wazuh-wui 4.0K Mar  2 20:43 .
drwxr-xr-x 9 wazuh-wui wazuh-wui 4.0K Apr 12 17:03 ..
-rwxr-xr-x 1 wazuh-wui wazuh-wui  835 Jan 13 02:07 kibana
-rwxr-xr-x 1 wazuh-wui wazuh-wui  776 Jan 13 02:07 kibana-keystore
-rwxr-xr-x 1 wazuh-wui wazuh-wui  813 Jan 13 02:07 kibana-plugin
sudo -u wazuh-wui bin/kibana-plugin install https://packages.wazuh.com/4.x/ui/kibana/wazuh_kibana-4.1.4_7.10.2-1.zip
sudo: bin/kibana-plugin: command not found
make[1]: *** [override_dh_install] Error 1
make[1]: Leaving directory `/build_wazuh/wui/wazuh-wui-5.0.0'
make: *** [binary] Error 2
dpkg-buildpackage: error: sudo debian/rules binary gave error exit status 2
debuild: fatal error at line 1357:
```
It seems that it is not possible to chang the directory while building the package.

**Comment by DFolchA** [comment]
## Update report 

Investigate and fix problem with services not being stopped on uninstall: 00f766c309a98fe0d18913bd0c38dc763c6952a3

Change services and build wazuh-wui package using 5.0.0 plugin.

Change login screen for wazuh-wui: 8067f9cd116f00630b4bb994243c24c6e482f532

![image](https://user-images.githubusercontent.com/19505384/115046824-312ba300-9ed8-11eb-9f75-f23933eaf306.png)

Test new changes.
